docs/markdown/Build-options.md

```diff
@@ -16,32 +16,37 @@ Here is a simple option file.
 option('someoption', type : 'string', value : 'optval', description : 'An option')
 option('other_one', type : 'boolean', value : false)
 option('combo_opt', type : 'combo', choices : ['one', 'two', 'three'], value : 'three')
+option('free_array_opt', type : 'array', value : ['one', 'two'])
 option('array_opt', type : 'array', choices : ['one', 'two', 'three'], value : ['one', 'two'])
 ```
 
-All types allow a `description` value to be set describing the option, if no
-option is set then the name of the option will be used instead.
+All types allow a `description` value to be set describing the option,
+if no option is set then the name of the option will be used instead.
 
 ### Strings
 
-The string type is a free form string. If the default value is not set then an
-empty string will be used as the default.
+The string type is a free form string. If the default value is not set
+then an empty string will be used as the default.
 
 ### Booleans
 
-Booleans may have values of either `true` or `false`. If not default value is
-supplied then `true` will be used as the default.
+Booleans may have values of either `true` or `false`. If not default
+value is supplied then `true` will be used as the default.
 
 ### Combos
 
-A combo allows any one of the values in the `choices` parameter to be selected.
-If no default value is set then the first value will be the default.
+A combo allows any one of the values in the `choices` parameter to be
+selected.  If no default value is set then the first value will be the
+default.
 
 ### Arrays
 
-Arrays allow one or more of the values in the `choices` parameter to be selected.
-If the `value` parameter is unset then the values of `choices` will be used as
-the default.
+Arrays represent an array of strings. By default the array can contain
+arbitrary strings. To limit the possible values that can used set the
+`choices` parameter. Meson will then only allow the value array to
+contain strings that are in the given list. The `value` parameter
```

Comment:
It might be good to mention that the array can still be empty if `choices` are provided.